src/sst/elements/merlin/topology/polarstar.h

```diff
@@ -0,0 +1,252 @@
+// SPDX-License-Identifier: MIT License
+
+// Author: Kartik Lakhotia
+
+#ifndef COMPONENTS_MERLIN_TOPOLOGY_POLARSTAR_H
+#define COMPONENTS_MERLIN_TOPOLOGY_POLARSTAR_H
+
+#include <sst/core/event.h>
+#include <sst/core/link.h>
+#include <sst/core/params.h>
+#include <sst/core/rng/rng.h>
+
+#include <string.h>
+#include <unistd.h>
+#include <vector>
+#include <algorithm>
+#include <stdlib.h>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+#include "sst/elements/merlin/router.h"
+
+namespace SST {
+namespace Merlin {
+
+
+//For now, polarstar_event is not much different from the router event. Just added the hop_count metric to record the total number of hops this packet is taking across the network
+class topo_polarstar_event : public internal_router_event {
+public:
+
+
+    int hop_count; //Count no of hops this packet has taken so far
+     
+    bool non_minimal;
+    int valiant;
+
+    topo_polarstar_event() {}
+    topo_polarstar_event(int hcount) : non_minimal(false) {	
+        
+        hop_count = hcount;
+        
+     }
+    virtual ~topo_polarstar_event() {}
+
+    virtual internal_router_event* clone(void) override
+    {
+        return new topo_polarstar_event(*this);
+         
+    }
+
+    void serialize_order(SST::Core::Serialization::serializer &ser)  override {
+        internal_router_event::serialize_order(ser);
+        
+        ser & hop_count;
+        ser & non_minimal;
+        ser & valiant;
+    }
+
+protected:
+
+private:
+    ImplementSerializable(SST::Merlin::topo_polarstar_event)
+
+};
+
+
+class topo_polarstar_init_event : public topo_polarstar_event {
+public:
+    //Insert code
+    int phase;
+
+    //track what is covered
+    int total_routers;
+    std::vector<int> covered;
+
+    topo_polarstar_init_event() {}
+    topo_polarstar_init_event(int p, int N) : topo_polarstar_event(0), phase(p), total_routers(N) { covered.resize(N, 0);}
+    virtual ~topo_polarstar_init_event() { }
+    virtual internal_router_event* clone(void) override
+    {
+        topo_polarstar_init_event* tte = new topo_polarstar_init_event(*this);
+        //Insert code
+        return tte;
+    }
+
+    void serialize_order(SST::Core::Serialization::serializer &ser)  override {
+        topo_polarstar_event::serialize_order(ser);
+        ser & phase;
+        ser & total_routers; 
+        for (int i=0; i<total_routers; i++)
+            ser & covered[i];
+         
+    }
+
+private:
+    ImplementSerializable(SST::Merlin::topo_polarstar_init_event)
+
+};
+
+
+class topo_polarstar: public Topology {
+
+public:
+
+    SST_ELI_REGISTER_SUBCOMPONENT_DERIVED(
+        topo_polarstar,
+        "merlin",
+        "polarstar",
+        SST_ELI_ELEMENT_VERSION(1,0,0),
+        "PolarStar: Highly Scalable Diameter-3 Topology Object",
+        SST::Merlin::Topology)
+
+    SST_ELI_DOCUMENT_PARAMS(
+        // Parameters needed for use with old merlin python module
+        {"polarstar.d", "Network Radix per router"},
```

Comment:
The polarstar.* params are not needed.  I'm sure you copied this from existing topologies, but these are in those topologies for backward compatibility to an older python module implementation.  They won't be needed for this one.

Reply:
> In addition to addressing the comments inlined in the files changed listing, please rebase against the latest devel branch of the element repo. We had a recent bug slip through testing that was recently fixed, and you will need to make sure you have that fix or you won't pass autotesting.

Should we create a pull request for the devel branch or the master?

Reply:
Please create the pull request to devel.
Documentation here may be useful: 
https://github.com/sstsimulator/sst-elements/blob/master/CONTRIBUTING.md
Thanks

Reply:
We are getting an error with SST_ELI_REGISTER_SUBCOMPONENT while compiling Merlin in the devel branch:

In file included from merlin.cc:25:
hr_router/xbar_arb_rr.h: At global scope:
hr_router/xbar_arb_rr.h:40:9: error: expected identifier before string constant
   40 |         "merlin",
      |         ^~~~~~~~
hr_router/xbar_arb_rr.h:40:9: error: expected ‘,’ or ‘...’ before string constant
hr_router/xbar_arb_rr.h:38:5: error: ISO C++ forbids declaration of ‘SST_ELI_REGISTER_SUBCOMPONENT’ with no type [-fpermissive]
   38 |     SST_ELI_REGISTER_SUBCOMPONENT(
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hr_router/xbar_arb_rr.h:45:5: error: expected ‘;’ at end of member declaration
   45 |     )
      |     ^
      |      ;